src/ed_infrastructure/utils/jwt/jwt_handler.py

```
@@ -0,0 +1,46 @@
+            return jwt.decode(
+                token,
+                self._secret,
+                algorithms=[self._algorithm],
+            )
+        except Exception as e:
+            LOG.error(f"Error decoding jwt token: {token}. {e}")
+            raise ApplicationException(
+                Exceptions.BadRequestException,
+                "Token validation failed.",
+                ["Token is malformed."],
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)